### ISSUE TITLE
onedatastore resource type ignores basepath parameter; breaks idempotence

### ISSUE THREAD

**Issue by mattthias** [epost-dev/opennebula-puppet-module#193]
The onedatastore resource type ignores the basepath parameter when creating the datastore. The datastore will be created with a default path. When running puppet a second time puppet tries to change the basepath which fails as the provider doesn't support that action.

How to reproduce: 
- Install centos6 machine
- install puppet 3.7.2 
- clone the opennebula-puppet-module and the puppetlabs-stdlib repo to a directory called "modules"
```bash
[root@centos6 ~]# mkdir modules
[root@centos6 ~]# git clone https://github.com/epost-dev/opennebula-puppet-module.git  modules/one
....
[root@centos6 ~]# git clone https://github.com/puppetlabs/puppetlabs-stdlib.git modules/stdlib
....
```
- create a simple site.pp
```puppet
node 'default' {

  class { one: 
    oned => true,
  }
  
  onedatastore { 'beispiel':
    ensure   => present,
    type     => 'IMAGE_DS',
    dm       => 'fs',
    tm       => 'ssh',
    basepath => '/srv/beispiel',
  }
}
```
- run puppet apply for the first time  (should work)
```bash
[root@centos6 ~]# puppet apply --modulepath=modules site.pp
...

```

- check if the datastore exists; check the path in the xml output. As we can see there is no basepath "srv/beispiel".
```bash
[root@centos6 ~]# onedatastore list 
  ID NAME                SIZE AVAIL CLUSTER      IMAGES TYPE DS      TM      STAT
   0 system                0M -     -                 0 sys  -       shared  on  
   1 default            29.4G 89%   -                 0 img  fs      shared  on  
   2 files              29.4G 89%   -                 0 fil  fs      ssh     on  
 100 beispiel           29.4G 89%   -                 0 img  fs      ssh     on  
[root@centos6 ~]# onedatastore list  -x | grep BASE
    <BASE_PATH><![CDATA[/var/lib/one//datastores/0]]></BASE_PATH>
      <BASE_PATH><![CDATA[/var/lib/one//datastores/]]></BASE_PATH>
    <BASE_PATH><![CDATA[/var/lib/one//datastores/1]]></BASE_PATH>
      <BASE_PATH><![CDATA[/var/lib/one//datastores/]]></BASE_PATH>
    <BASE_PATH><![CDATA[/var/lib/one//datastores/2]]></BASE_PATH>
      <BASE_PATH><![CDATA[/var/lib/one//datastores/]]></BASE_PATH>
    <BASE_PATH><![CDATA[/var/lib/one//datastores/100]]></BASE_PATH>
      <BASE_PATH><![CDATA[/var/lib/one//datastores/]]></BASE_PATH>
[root@centos6 ~]# 
```

- run puppet apply again (will surely break)
```bash
[root@centos6 ~]# puppet apply --modulepath=modules/ site.pp 
Notice: Compiled catalog for centos6 in environment production in 2.73 seconds
Error: Can not modify basepath. You need to delete and recreate the datastore
Error: /Stage[main]/Main/Node[default]/Onedatastore[beispiel]/basepath: change from /var/lib/one//datastores/ to /srv/beispiel failed: Can not modify basepath. You need to delete and recreate the datastore
Notice: Finished catalog run in 5.95 seconds
[root@centos6 ~]# 
```

**Comment by rwaffen** [comment]
will be fixed with #191 ...

**Comment by mattthias** [comment]
```bash
Error: Can not modify basepath. You need to delete and recreate the datastore
Error: /Stage[main]/Main/Node[default]/Onedatastore[beispiel]/base_path: change from /srv/beispiel/100 to /srv/beispiel failed: Can not modify basepath. You need to delete and recreate the datastore
```
